test/config.ts

```diff
@@ -64,7 +64,7 @@ setConfig(config)
 
 const options: PlaywrightOptions = {
   headless: true, // Run tests in headless browsers.
-  video: "retain-on-failure",
+  video: "retry-with-video",
```

Comment:
I'm actually not sure if we want `retain-on-failure` or `retry-with-video`. Here is the difference:

> `retain-on-failure` - Record video for each test, but remove all videos from successful test runs.
> `retry-with-video` - Record video only when retrying a test.

If we do the `retain-on-failure` option, we only have videos for failures. If we do the other option, we keep the videos from test retries. And a test is only retried if it fails. Hmm..thinking out loud, I think we want to keep it on `retain-on-failure`. Open to other ideas though!

Source: [`playwright-test` docs](https://github.com/microsoft/playwright-test#modify-options)

Reply:
Is there a way to just get it to always keep videos, regardless of pass/fail/retry? They are useful to make sure our tests are doing what we think they're doing -- otherwise, we can have tests that pass but are not testing what we think they're testing (I've certainly written my share of tests that were not testing what I thought I was testing! and the passing tests gave me a false sense of security).

I know that disk usage can be a concern, but we can address that through other means, such as a shorter retention duration for old tests?

Reply:
Yup! According to the docs we can use `on` here to always record videos.

> I know that disk usage can be a concern, but we can address that through other means, such as a shorter retention duration for old tests

Now you're asking for extra fancy features! That's going to cost ya! 😛

AFAIK, that isn't an option built-in to playwright and would probably be something we run separately (maybe a GitHub Action) on a schedule. I think we can address that if/when it comes up. 

For now, I'll change it to `on`.

Reply:
Oh yeah I meant GitHub has a setting for retention in our admin panel!

![image](https://user-images.githubusercontent.com/52710/115060285-8944b400-9e9c-11eb-8d6c-02ddddca2b27.png)

If it becomes a problem we can move the builds to Google Cloud and store them in Cloud Storage, too

Reply:
Sorry, it would've been better if I linked you to it lol https://github.com/cdr/code-server/settings/actions

Reply:
Ohhh!!!!!!!!! 

TIL. 

You already know how to fix/address then 😅 We're good then.